### PR TITLE
Make 1.1 API netcoreapp1.1-specific

### PR DESCRIPTION
**Pull request by ericstj**
API was previously added to InteropServices and RegEx as `netstandard1.6`.

Both of these packages are part of NETStandard.Library so placing the API in a netstandard folder commits that API to the standard.  Since this is brand-new API that is not yet implemented anywhere but .NET Core (and not even baked) we shouldn't be committing it to the standard.

This change moves the new API to a netcoreapp1.1 specific reference assembly.

Replaces https://github.com/dotnet/corefx/pull/11272.

/cc @weshaggard

### CODE REVIEW

**Comment by davidfowl** [comment]
YES!

**Comment by ericstj** [comment]
Looks like the tests need fixing...  Will follow up.

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test

File: src/System.Linq.Expressions/tests/System.Linq.Expressions.Tests.csproj [hunk 1]
```diff
@@ -226,6 +226,8 @@
       <Project>{aef718e9-d4fc-418f-a7ae-ed6b2c7b3787}</Project>
       <Name>System.Linq.Expressions</Name>
     </ProjectReference>
+    <!-- temporarily compile against the live build of RegularExpressions to handle assembly-version change -->
+    <ProjectReference Include="..\..\System.Text.RegularExpressions\pkg\System.Text.RegularExpressions.pkgproj"/>
```
Comment (weshaggard) [inline src/System.Linq.Expressions/tests/System.Linq.Expressions.Tests.csproj]: Can you please be sure to have a issue tracking the update of this in the release branch?

Comment (ericstj) [inline src/System.Linq.Expressions/tests/System.Linq.Expressions.Tests.csproj]: Sure.

Comment (ericstj) [inline src/System.Linq.Expressions/tests/System.Linq.Expressions.Tests.csproj]: https://github.com/dotnet/corefx/issues/11916

**Pull request merged**